modules/behavior/draw_way.js

```diff
@@ -395,6 +399,9 @@ export function behaviorDrawWay(context, wayID, mode, startGraph) {
             return;
         }
 
+        _secondLastAddedNode = _lastAddedNode;
```

Comment:
I've noticed two things here:
1) The behavior is a bit buggy when we use it in combination with undo/redo. Repro: Draw 2 nodes -> F -> Ctrl+Z -> F
2) If the first node that is placed is an existing one, then we actually need 3 vertexes in order to get started...

Allow me to propose the following: Instead of persisting the _last and _secondLast, try to derive this information from the current way you are modifying (_origWay,nodes). This state should follow editting history by default.
If this works its a 1min fix and would nail both 1) and 2).

Reply:
thanks, your suggestion works very smoothly and fixes both these issues